src/osd/PGBackend.cc

```diff
@@ -447,7 +447,7 @@ enum scrub_error_type PGBackend::be_compare_scrub_objects(
 		  << (known ? "known" : "best guess")
 		  << " omap_digest 0x" << auth.omap_digest << std::dec
 		  << " from auth shard " << auth_shard;
-      result.set_omap_digest_mismatch();
+      result.set_omap_digest_mismatch(known);
```

Comment:
@athanatos Here is where we have `bool know` which indicates whether we are guessing that this is the bad shard or not.  If `know` is true then we mark the shard and object with the error flag.  If `know` is false we only mark the object with the error flag by using `guessed`.